Source/ARTDefault.m

```diff
@@ -109,4 +136,54 @@ + (NSString *)bundleBuildNumber {
     return infoDictionary[ARTDefault_bundleBuildNumberKey];
 }
 
++ (NSString *)osName {
+    return
+        #if TARGET_OS_IOS
+            @"iOS"
+        #elif TARGET_OS_TV
+            @"tvOS"
+        #elif TARGET_OS_WATCH
+            @"watchOS"
+        #elif TARGET_OS_OSX
+            @"macOS"
+        #else
+            @"Unknown"
```

Comment:
Hi @maratal - sorry for the late spot on this one but I think this is a problem. We're going to end up with `Unknown/x.y.z` arriving at our servers and that carries little context or value.

I would prefer that this method returns `nil` if the OS cannot be identified, meaning that the `agent` method will need to be refactored so that it doesn't attempt to pack OS name or version in that scenario.

Reply:
https://github.com/ably/ably-cocoa/pull/1119/commits/68b1db88db1ce4b54ec9f2715a5761e87c2aa12a